x-pack/plugins/apm/common/utils/get_preferred_bucket_size_and_data_source.ts

```diff
@@ -28,13 +28,13 @@ export function getPreferredBucketSizeAndDataSource({
   sources,
   bucketSizeInSeconds,
 }: {
-  sources: ApmDataSource[];
+  sources: ApmDataSourceWithSummary[];
```

Comment:
if we add `hasDocs` to this, we also have to filter out all sources where `hasDocs = false`

Reply:
I'm not too comfortable depending on the consumer knowing that, giving `hasDocs` is present.